src/enzyme_ad/jax/Passes/EnzymeHLOOpt.cpp

```diff
@@ -2153,6 +2153,50 @@ struct ConcatToBroadcast final
   }
 };
 
+struct DynamicUpdateSliceConstProp final
+    : OpRewritePattern<mlir::stablehlo::DynamicUpdateSliceOp> {
+  using OpRewritePattern::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(mlir::stablehlo::DynamicUpdateSliceOp op,
+                                PatternRewriter &rewriter) const override {
+    auto startIndices = op.getStartIndices();
+
+    bool legal = true;
+
+    DenseElementsAttr operandConstant;
+    DenseElementsAttr updateConstant;
+
+    SmallVector<DenseElementsAttr> constants(startIndices.size(),
+                                             DenseElementsAttr());
+    for (auto &operand : op->getOpOperands()) {
+      if (operand.getOperandNumber() == 0)
+        legal &= matchPattern(operand.get(), m_Constant(&operandConstant));
+      else if (operand.getOperandNumber() == 1)
+        legal &= matchPattern(operand.get(), m_Constant(&updateConstant));
+      else
+        legal &= matchPattern(
+            operand.get(),
+            m_Constant(&constants[operand.getOperandNumber() - 2]));
+    }
+
+    if (!legal)
+      return failure();
+
```

Comment:
Minor but can you add a special case if both input and update are splats of the same value and to return a resized splat.

the reason for this is if you have a giant op otherwise we will iterate through all points in tensor and slow things down a lot